It looks like there are some tests failing
```
17:41:49 ======================================================================
17:41:49 FAIL: test_sparse_default_std (test_nn.TestNNInit)
17:41:49 ----------------------------------------------------------------------
17:41:49 Traceback (most recent call last):
17:41:49   File "test_nn.py", line 5124, in test_sparse_default_std
17:41:49     assert column[column == 0].nelement() >= math.ceil(sparsity * cols)
17:41:49 AssertionError
17:41:49 
17:41:49 ----------------------------------------------------------------------
```

By [looking at the failing test](https://github.com/pytorch/pytorch/blob/master/test/test_nn.py#L5179), I think that there is a bug in the sparsity check, and it should be `math.ceil(sparsity * rows)` instead, as in [the `sparse` implementation](https://github.com/pytorch/pytorch/blob/master/torch/nn/init.py#L363)

Could you verify that this is indeed the case, and fix the test?

--- a/test/test_nn.py
+++ b/test/test_nn.py
@@ -5121,8 +5121,7 @@ class TestNNInit(TestCase):
 
                 for col_idx in range(input_tensor.size(1)):
                     column = input_tensor[:, col_idx]
-                    assert column[column == 0].nelement() >= math.ceil(sparsity * cols), 
-                           "{} : {}, {}".format(column[column == 0].nelement(), sparsity, cols)
+                    assert column[column == 0].nelement() >= math.ceil(sparsity * cols), "{} : {}, {}".format(column[column == 0].nelement(), sparsity, cols)
 
                 assert self._is_normal(input_tensor[input_tensor != 0], 0, std)
 

--- a/test/test_nn.py
+++ b/test/test_nn.py
@@ -5121,7 +5121,8 @@ class TestNNInit(TestCase):
 
                 for col_idx in range(input_tensor.size(1)):
                     column = input_tensor[:, col_idx]
-                    assert column[column == 0].nelement() >= math.ceil(sparsity * cols), "{} : {}".format(column[column == 0].nelement(), math.ceil(sparsity * cols))
+                    assert column[column == 0].nelement() >= math.ceil(sparsity * cols), 
+                           "{} : {}, {}".format(column[column == 0].nelement(), sparsity, cols)
 
                 assert self._is_normal(input_tensor[input_tensor != 0], 0, std)
 

--- a/test/test_nn.py
+++ b/test/test_nn.py
@@ -5121,7 +5121,7 @@ class TestNNInit(TestCase):
 
                 for col_idx in range(input_tensor.size(1)):
                     column = input_tensor[:, col_idx]
-                    assert column[column == 0].nelement() >= math.ceil(sparsity * cols)
+                    assert column[column == 0].nelement() >= math.ceil(sparsity * cols), "{} : {}".format(column[column == 0].nelement(), math.ceil(sparsity * cols))
 
                 assert self._is_normal(input_tensor[input_tensor != 0], 0, std)
 

--- a/torch/nn/init.py
+++ b/torch/nn/init.py
@@ -360,7 +360,7 @@ def sparse_(tensor, sparsity, std=0.01):
         raise ValueError("Only tensors with 2 dimensions are supported")
 
     rows, cols = tensor.shape
-    num_zeros = int(math.ceil(rows * sparsity))
+    num_zeros = int(math.ceil(sparsity * rows))
 
     with torch.no_grad():
         tensor.normal_(0, std)

